CONTRIBUTING.md

```diff
@@ -29,3 +29,138 @@ Recommendations and requirements for how to best contribute to Parquet. We striv
 ### License
 By contributing your code, you agree to license your contribution under the terms of the APLv2:
 https://github.com/apache/parquet-format/blob/master/LICENSE
+
+### Additions/Changes to the Format
+
+Note: This section applies to actual functional changes to the specification.
+Fixing typos, grammar, and clarifying concepts that would not change the
+semantics of the specification can be done as long a comitter feels comfortable
+to merge them. When in doubt starting a discussion on the dev mailing list is
+encouraged.
+
+The general steps for adding features to the format are as follows:
+
+1. Discuss changes on the developer mailing list (dev@parquet.apache.org).
+   Often times it is helpful to link to a draft pull request to make the
+   discussion concrete. This step is complete when there is lazy consensus. Part
+   of the consensus is whether it sufficient to provide 2 working
+   implementations as outlined in step 2 or if demonstration of the feature with
+   a down-stream query engine is necessary to justify the feature (e.g.
+   demonstrate performance improvements in Arrow's DataSet library or Apache
+   Data Fusion or another open source engine).
+
+2. Once a change has lazy consensus, two implementations of the feature
+   demonstrating interopability must also be provided.  One implementation MUST
+   be [parquet-java](http://github.com/apache/parquet-java).  It is preferred
+   that the second implementation be
+   [parquet-cpp](https://github.com/apache/arrow) or
+   [parquet-rs](https://github.com/apache/arrow-rs), however at the discretion
+   of the PMC any open source Parquet implementation may be acceptable.
+   Implementations whose contributors actively participate in the community
+   (e.g. keep their feature matrix up-to-date on parquet-site) are more likely
+   to be considered. If discussed as a requirement in step one, demonstration
+   of integration with a query engine is also required for this step. The
+   implementations must be made available publicly (e.g. as a pull request
+   against the target repository).
```

Comment:
We might be more specific here to avoid giving the impression that a "code dump" is sufficient to get a format change to be accepted, though perhaps I'm overly cautious :-)
```suggestion
   of integration with a query engine is also required for this step. The
   implementations must be made available publicly, and they should
   be fit for inclusion (for example, they were submitted as a pull request
   against the target repository and committers gave positive reviews).
```